homeworks/Yaroslav.Belinskyi_YaroslavBelinskyi/homework_5/firstscenario.js

```diff
@@ -0,0 +1,68 @@
+// 1 task
+const exampleArray = [10, -10, 10, -10, 10];
+const resultArray = [];
+function sumArray(arr) {
+    arr.reduce((previousSum, currentNumber) => {
+        const value = previousSum + currentNumber;
+        resultArray.push(value);
+        return value;
+    }, 0);
+}
+sumArray(exampleArray);
+console.log(resultArray);
+// 2 task, please write a hint if there are any easier method for it
+const classes = ['header', 'menu', 'menu_item', 'tabs',
+    'tab_item', 'menu', 'link', 'tabs', 'tab_item', 'menu',
+    'menu_item', 'menu', 'menu_item'];
+const obj = {};
+function repeatsInObj(array) {
+    array.forEach((element) => {
+        obj[element] = 0;
+    });
+    const objKeys = Object.keys(obj);
+    array.forEach((element) => {
+        for (let i = 0; i < objKeys.length; i++) {
+            if (objKeys[i] === element) {
+                const key = objKeys[i];
+                obj[key] += 1;
+            }
+        }
+    });
+    console.log(obj);
+}
+repeatsInObj(classes);
+// 3 task, I know this the easiest way to copy everything :D
+const objForCopy = { b: 'f', d: { e: 'f' } };
+function innerCopy(object) {
+    const objParsed = JSON.stringify(object);
+    const objCopy = JSON.parse(objParsed);
+    return objCopy;
+}
+const objInnerCopy = innerCopy(objForCopy);
```

Comment:
nope, needs to be solved in different way, not using JSON methods

Reply:
done